tutorial/04_embedding_net.ipynb

```diff
@@ -0,0 +1,385 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Learning summary features with a neural net\n",
+    "\n",
+    "When doing simulation-based inference, it is very important to use well-chosen summary statistics for describing the data generated by the simulator. Very often, these statistics take into account previous domain knowledge. For instance, in the case of the Hodgkin-Huxley model from this [tutorial](https://www.mackelab.org/sbi/tutorial/02_HH_simulator/), the summary statistics are defined via the function defined in [here](https://github.com/mackelab/sbi/blob/bc3e6f2473bf5e61ad844e975f68b462d4d64dd8/tutorial/HH_helper_functions.py#L159), which takes a 120 ms recording as input (a 12000-dimensional input vector) and outputs a 7-dimensional feature vector containing different statistical descriptors of the recording (e.g., number of spikes, average value, etc.). However, in some occasions, it might be of interest to actually **learn from the data** which summary statistics to use. For this, one may resort to different learning strategies, such as Gaussian processes, kernel density estimates, and neural networks.\n",
+    "\n",
+    "In the `sbi` package, we may instantiate a learnable summary statistics layer (referred as `embedding_net`) and use it with the neural density estimator. The `embedding_net` serves as an input transform layer for this neural posterior and their parameters are optimized together during training.\n",
```

Comment:
If I may propose an alternative formulation these two sentences:

`sbi` offers functionality to learn summary statistics from (potentially high-dimensional) simulation outputs with a neural network. In `sbi`, this neural network is referred to as `embedding_net`. If an `embedding_net` is specified, the simulation outputs are passed through the `embedding_net`, whose outputs are then passed to the neural density estimator. The parameters of the `embedding_net` are updated together with the parameters of the neural density estimator.

Reply:
After this paragraph, maybe add a new paragraph:

Note: only `SNPE` and `SNRE` methods can use an `embedding_net` to learn summary statistics from simulation outputs. `SNLE` does not offer this functionality since the simulation outputs $x$ are the outputs of the neural density estimator in `SNLE`.